presto-spi/src/main/java/com/facebook/presto/spi/security/AccessDeniedException.java

```diff
@@ -68,6 +68,11 @@ public static void denyRenameSchema(String schemaName, String newSchemaName, Str
         throw new AccessDeniedException(format("Cannot rename schema from %s to %s%s", schemaName, newSchemaName, formatExtraInfo(extraInfo)));
     }
 
+    public static void denyShowCatalog(String catalogName)
+    {
+        throw new AccessDeniedException(format("Cannot show catalog %s", catalogName));
```

Comment:
It might not be visible to user as 'MetadataManager.getCatalogNames` hides the unauthorized entry.